sources/api/netdog/src/wicked/vlan.rs

```diff
@@ -0,0 +1,16 @@
+use crate::interface_name::InterfaceName;
+use serde::Serialize;
+
+#[derive(Debug, Clone, Serialize, PartialEq)]
+pub(crate) struct WickedVlanTag {
+    #[serde(rename = "$unflatten=device")]
+    device: InterfaceName,
+    #[serde(rename = "$unflatten=tag")]
+    id: u16,
+}
+
+impl WickedVlanTag {
+    pub(crate) fn new(device: InterfaceName, id: u16) -> Option<Self> {
+        Some(WickedVlanTag { device, id })
+    }
```

Comment:
It's a bit unconventional to have a constructor return an `Option` - this should probably return a `Self` and it can be wrapped with `Some()` if needed elsewhere.

Reply:
Oh, yeah, now that you say that, that makes sense. I'll move it to where it should be.